sdks/java/io/redis/src/main/java/org/apache/beam/sdk/io/redis/RedisIO.java

```diff
@@ -260,16 +261,12 @@ public ReadAll withConnectionConfiguration(RedisConnectionConfiguration connecti
 
   }
 
-  /**
-   * A {@link DoFn} requesting Redis server to get key/value pairs.
-   */
-  private static class ReadFn extends DoFn<String, KV<String, String>> {
-
-    private final RedisConnectionConfiguration connectionConfiguration;
+  private abstract static class BaseReadFn<T> extends DoFn<String, T> {
+    protected final RedisConnectionConfiguration connectionConfiguration;
 
-    private transient Jedis jedis;
+    protected transient Jedis jedis;
 
-    public ReadFn(RedisConnectionConfiguration connectionConfiguration) {
+    public BaseReadFn(RedisConnectionConfiguration connectionConfiguration) {
```

Comment:
remove public, in general it is a common Beam practice to restrict access as much as possible.  You can use IntelliJ's analyze code to do this.